python/ray/_private/runtime_env/physical_mode_execution_utils.py

```diff
@@ -0,0 +1,177 @@
+"""A few util functions for physical mode execution.
+NOTE:
+1. It should be only used under linux environment; for other platforms, all functions
+are no-op.
+2. It assumes cgroup v2 has been mounted.
+"""
+# TODO(hjiang):
+# 1. Consider whether we need to support V1.
+# 2. Check the existence for cgroup V2.
+# 3. Add error handling; for now just use boolean return to differentiate success or
+# failure. Better to log the error.
+
+from ray._private.runtime_env import physical_mode_context
+
+import sys
+import os
+import shutil
+
+# There're two types of memory cgroup constraints:
+# 1. For those with limit memory capped, they will be created a dedicated cgroup;
+# 2. For those without limit memory specified, they will be added to the default memory
+# cgroup.
+DEFAULT_MEM_CGROUP_UUID = "default_cgroup_memory_uuid"
+
+
+def _setup_new_cgroup_for_memory(
+    physical_exec_ctx: physical_mode_context.PhysicalModeExecutionContext,
+) -> bool:
+    """Setup a new cgroup for memory consumption. Return whether memory constraint is
+    applied successfully."""
+    if not physical_exec_ctx.memory_cgroup_uuid:
+        return False
+
+    if physical_exec_ctx.memory_cgroup_uuid == DEFAULT_MEM_CGROUP_UUID:
+        return False
+
+    cgroup_folder = f"/sys/fs/cgroup/{physical_exec_ctx.memory_cgroup_uuid}"
```

Comment:
I think having a flat child structure may be problematic once we introduce cpu isolation, I'd prefer to be a tree structure.

Reply:
When I wrote this PR, I'm thinking of a flat structure, where we place all CPU / memory cgroup as leaf node, whether there's max / min specified.

Reply:
Discussed offline, will switch to tree structure.